src/components/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,113 @@
+import UserMenuBuddy from 'components/UserMenuBuddy';
+import { isBuddy, isNewbie } from 'utils';
+import theme from 'styles/theme';
+import { ROUTES } from 'shared/routes';
+import { Buddy, Newbie, UserRole } from 'buddy-app-schema';
+import AuthContext, { AuthContextData } from 'contexts/AuthContext';
+import { BasicDetailsParams, UserMenuProps, UserBasicDetails } from './types';
+
+const useStyles = makeStyles({
+  list: {
+    width: '35rem',
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  closeBtn: {
+    alignSelf: 'flex-end',
+    zIndex: theme.zIndex.base,
+  },
+  loader: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100vh',
+    width: '100%',
+  },
+});
+
+const UserMenu: React.FC<UserMenuProps> = props => {
+  const history = useHistory();
+  const { list, closeBtn, loader } = useStyles();
+  const { isMenuVisible, onClose } = props;
+  const { data: AuthData } = useContext<AuthContextData>(AuthContext);
+  const { userId, role } = AuthData;
+
+  const getQueryByRole = (role: UserRole, id: string) => {
+    if (isBuddy(role)) {
+      return { query: BUDDY_MENU_DETAILS, variables: { buddyId: id } };
+    } else if (isNewbie(role)) {
+      return { query: NEWBIE_MENU_DETAILS, variables: { newbieId: id } };
+    }
+  };
+
+  const selectNewbie = (id: string) => {
+    history.push(ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', id));
+    onClose();
+  };
+
+  const selectBuddy = (id: string) => {
+    history.push(ROUTES.NEWBIE_BUDDY_DETAILS.replace(':buddyId', id));
+    onClose();
+  };
+
+  const { query, variables } = getQueryByRole(role, userId) || {};
+  const { data, loading } = useQuery<UserBasicDetails, BasicDetailsParams>(query, {
+    variables,
+  });
+  const user = data && data[role.toLowerCase()];
+  return (
+    <Drawer open={isMenuVisible} onClose={onClose} data-testid='slide-menu'>
+      <Box className={list}>
+        {user && (
+          <Box data-testid='slide-menu-body'>
+            <Box display='flex' justifyContent='flex-end'>
+              <IconButton
+                className={closeBtn}
+                onClick={onClose}
+                data-testid='slide-menu-close-btn'>
+                <CloseIcon />
+              </IconButton>
+            </Box>
+            <UserMenuDetails user={user} />
+            <Divider />
+            {isBuddy(role) && (
+              <UserMenuNewbies
+                newbies={(user as Buddy).newbies}
+                onSelect={selectNewbie}
+              />
+            )}
+            {isNewbie(role) && (
+              <UserMenuBuddy buddy={(user as Newbie).buddy} onSelect={selectBuddy} />
+            )}
+            <Divider />
+            <UserMenuSettings
+              allowPushedNotifications={!!user.allowPushedNotifications}
+              updatePushNotificationsSettings={() => {}}
+            />
+          </Box>
+        )}
+        {loading && (
```

Comment:
I totally see the usage of Suspense in this component. 
It can just be lazy-loaded when the user enters in newbieSelect screen and not only when the user clicks on the navBar button.

Reply:
Sounds like good candidate for a performance optimisation task in our app. I've never used suspense before, but I guess we can create separate task for that as an improvement.

Reply:
Sure @michaelmoney can you please create a US in our backlog to tackle this?

Reply:
Done. If you would like to add sth, feel free:
https://digitalrig.atlassian.net/browse/BUD-207